include/wil/wistd_functional.h

```diff
@@ -297,13 +297,19 @@ namespace wistd     // ("Windows Implementation" std)
       template <class _Fp>
       using _EnableIfCallable = typename enable_if<__callable<_Fp>::value>::type;
     public:
-        typedef _Rp result_type;
+        using result_type = _Rp;
 
         // construct/copy/destroy:
         __WI_LIBCPP_INLINE_VISIBILITY __WI_LIBCPP_SUPPRESS_NONINIT_ANALYSIS
-        function() WI_NOEXCEPT : __f_(0) {}
+        function() WI_NOEXCEPT : __buf_(), __f_(nullptr)

```

Comment:
Initializing the buffer (here and below) seems odd to me. You don't zero initialize the memory returned by `operator new` before calling placement `new`, so there's no real benefit here